03-Binary_data_to_computations.Rmd

```diff
@@ -102,12 +102,12 @@ Examples of commonly used operating systems on computers and phones are:
 * Linux  
 * Android
 
-Recall that we previously talked about how computers today are often called 64-bit? Operating systems are also designed in this way. A 64-bit operating system expects the hardware of the computer to allow for processing at least 64 bits of data at a time (the word size) [@Wikipedia_word_length_2021]. If we have registers of at least this length in the CPU, than we can in fact perform operations on data that may be up to 64 bits in length. This also means that we can perform operations on values that take up less than 64 bits. This can be important because if you try to use an operating system that expects a longer word size than the hardware can accommodate, for example a 64-bit operating system on a 32-bit computer, this will not work. Application programs are also designed according to different word sizes and again you need to choose options that are equal to or less than the word size that your CPU can accommodate [@ComputerHope_64-bit].
+Recall that we previously talked about how computers today are often called 64-bit? Operating systems are also designed in this way. A 64-bit operating system expects the hardware of the computer to allow for processing at least 64 bits of data at a time (the word size) [@Wikipedia_word_length_2021]. If we have registers of at least this length in the CPU, then we can in fact perform operations on data that may be up to 64 bits in length. The data do not __have__ to be the full 64 bits; it just means that we can perform operations on values that take up less than 64 bits. This can be important because if you try to use an operating system that expects a longer data size than the hardware can accommodate, for example a 64-bit operating system on a 32-bit computer, this will not work. Application programs are also designed according to different data sizes and again you need to choose options that are equal to or less than the data size that your CPU can accommodate [@ComputerHope_64-bit]. However, you can run a 32-bit operating system on a 64-bit computer, and a 32-bit application on a 64-bit operating system.
 
```

Comment:
AVOCADO: I substituted “data sizes” for “word sizes” referring to 64- and 32-bits, is this appropriate? Also, is the last sentence correct? This was my understanding from the paragraph, and I thought it would be nice to give an illustrative sentence here. (Last sentence of line 105)

Reply:
I believe that word size is more appropriate. However maybe we need to emphasize the definition of word size earlier in the paragraph more. Yes, the last sentence is correct, and probably good to include. I wonder if we should point out however that there may be reduced efficiency if a 32-bit application is used when a 64- bit application exists.  - could point to this [article](https://medium.com/codixlab/what-happens-when-a-32-bit-program-runs-on-a-64-bit-machine-c231ac3ddb2f) about this: